### DISCUSSION TITLE
Proper way to handle pull-requests?

### DISCUSSION THREAD

**Discussion by gaedep59197** [concourse/concourse#7210] [Help & Support] [answered]
We've recently starting using Concourse for our Continuous Integration. Currently we're using it to evaluate pul-requests from Bitbucket and run some tests on them. I'm using https://github.com/zarplata/concourse-git-bitbucket-pr-resource as the resource type and use the "version: every" tag as suggested. I've also been reading quite a bit and have been looking at the v10 roadmap. It seems to deal with some of my concerns on dealing with pull-requests as well, which is great to see. I've also seen there have been attempts to do forsee certain things already, like `across` and `set-pipeline`. I don't think I fully grok these concepts yet, but would be willing to experiment with them if they can be used for a better way to handle pull-requests?

I guess my question boils down to: What is, currently, the best way to handle pull requests?

**Comment by taylorsilva** [comment, marked as answer]
Sounds like you already have a similar setup to what we have. You can see the Concourse PR's pipeline here: https://github.com/concourse/ci/blob/master/pipelines/pr.yml
There's one PR pipeline per branch: https://ci.concourse-ci.org/?search=team%3A%22contributor%22%20group%3A%22pr%22

Ideally in the future we can use the `across` + `set_pipeline` steps to set a PR pipeline per PR submitted.

**Reply by gaedep59197** [reply]
Ok, that's what I thought. I've been perusing the ci repo of concourse a bit and though I do see some uses of `across` and `set_pipeline`, it's doesn't seem like it's capable of doing what I'd like to do, but that's ok! What we have now works well enough!

I might be tempted to learn some Go and see if I can't help on that front, but we'll see. Too much to do and too little time :)

**Reply by aoldershaw** [reply]
@gaedep59197 with the coming release 7.4, PR workflows using `across` and `set_pipeline` will be possible. We're currently running it on our CI at https://ci.concourse-ci.org/?search=team%3A%22contributor%22%20group%3A%22pr%22

(note that not all of the PR pipelines are marked as exposed currently, so there's only a subset of them visible if you aren't logged in and on the `contributors` team)

Take a look at https://github.com/concourse/ci/blob/ecdae4873b7/pipelines/reconfigure.yml#L437-L461 for the job that's setting these PR pipelines
